streaming/base/dataset.py

```diff
@@ -804,12 +804,27 @@ def load_state_dict(self, obj: Dict[str, Any]) -> None:
         Args:
             obj (Dict[str, Any]): The state.
         """
+        # Set shared memory block to be 1024 characters long. This enables calling
+        # `load_state_dict` multiple times without needing to resize the shared memory block.
+        # Resizing the shared memory block is not possible, and closing the shared memory block
+        # and replacing it with a new one is causing great difficulties.
+
         name = _get_path(self._shm_prefix_int, RESUME)
-        data = json.dumps(obj, sort_keys=True).encode('utf-8')
+        data = json.dumps(obj, sort_keys=True)
+
+        len_needed = len(data)
+        if len_needed > RESUME_SHM_SIZE:
+            raise ValueError(f'State dict for resumption is too large to fit in shared memory. ' +
+                             f'Please reduce the state dict size to less than {RESUME_SHM_SIZE} ' +
+                             f'bytes. Current size is {len_needed} bytes.')
         # Some platforms choose to allocate chunks of memory based upon that platform's memory page
         # size, hence the exact size of the shared memory block that was returned may be larger
         # than what was requested.
-        self._resume_shm = SharedMemory(name=name, size=len(data))
+        self._resume_shm = SharedMemory(name=name, size=RESUME_SHM_SIZE)
+        # Write a null byte at the end of the shared memory block so that we read in the state
+        # dict correctly in `_resume`.
+        data += '\0'
```

Comment:
Pedantic nit: += is linear time because str is immutable, right?

```
shm[:len(data)] = data
shm[len(data)] = 0
```

Reply:
Tried it, this throws an error: `memoryview: invalid type for format 'B'` :(
I tried encoding the null byte separately to utf-8, and adding that in as well, to no avail. So just went with this.

Reply:
Example:
```>>> string = "hello"
>>> stringbytes = string.encode('utf-8')
>>> mem.buf[:len(string)] = stringbytes
>>> mem.buf[len(string)] = '\0'.encode('utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: memoryview: invalid type for format 'B'
```